tests/UI/campaigns/data/faker/cartRule.js

```diff
@@ -2,12 +2,47 @@ const faker = require('faker');
 
 module.exports = class CartRule {
   constructor(cartRuleToCreate = {}) {
+    // Information
     this.name = cartRuleToCreate.name || faker.commerce.department();
     this.description = faker.lorem.sentence();
     this.code = cartRuleToCreate.code;
-    this.customer = cartRuleToCreate.customer || faker.internet.email();
+    this.highlight = cartRuleToCreate.highlight === undefined ? false : cartRuleToCreate.highlight;
+    this.partialUse = cartRuleToCreate.partialUse === undefined ? true : cartRuleToCreate.partialUse;
+    this.priority = cartRuleToCreate.priority || 1;
+    this.status = cartRuleToCreate.status === undefined ? true : cartRuleToCreate.status;
+
+    // Conditions
+    this.customer = cartRuleToCreate.customer || false;
+    this.dateFrom = cartRuleToCreate.dateFrom || false;
+    this.dateTo = cartRuleToCreate.dateTo || false;
+
+    this.minimumAmount = {
+      value: cartRuleToCreate.minimumAmount === undefined ? 0 : cartRuleToCreate.minimumAmount.value,
+      currency: cartRuleToCreate.minimumAmount === undefined ? 'EUR' : cartRuleToCreate.minimumAmount.currency,
+      tax: cartRuleToCreate.minimumAmount === undefined ? 'Tax included' : cartRuleToCreate.minimumAmount.tax,
+      shipping: cartRuleToCreate.minimumAmount === undefined
+        ? 'Shipping included' : cartRuleToCreate.minimumAmount.shipping,
+    };
+
+    this.quantity = cartRuleToCreate.quantity || 1;
+    this.quantityPerUser = cartRuleToCreate.quantityPerUser || 1;
+
+    // Actions
     this.freeShipping = cartRuleToCreate.freeShipping === undefined ? 'on' : cartRuleToCreate.freeShipping;
-    this.percent = cartRuleToCreate.percent === undefined ? true : cartRuleToCreate.percent;
-    this.value = cartRuleToCreate.value || faker.random.number({min: 1, max: 100});
+
+    this.discountType = cartRuleToCreate.discountType || 'None';
+    if (this.discountType === 'Percent') {
+      this.discountPercent = cartRuleToCreate.discountPercent || faker.random.number({min: 10, max: 80});
```

Comment:
Why not 
```suggestion
      this.discountPercent = cartRuleToCreate.discountPercent || faker.random.number({min: 1, max: 99});
```

Reply:
Are you referring to this https://github.com/PrestaShop/PrestaShop/pull/21188#discussion_r496049902 ?

Reply:
ok